internal/repository/token_repository.go

```diff
@@ -0,0 +1,44 @@
+package repository
+
+import (
+	"context"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+
+	"example.com/tfgrid-kyc-service/internal/models"
+)
+
+type MongoTokenRepository struct {
+	collection *mongo.Collection
+}
+
+func NewMongoTokenRepository(db *mongo.Database) TokenRepository {
+	return &MongoTokenRepository{
+		collection: db.Collection("tokens"),
+	}
+}
+
+func (r *MongoTokenRepository) SaveToken(ctx context.Context, token *models.Token) error {
+	token.CreatedAt = time.Now()
+	_, err := r.collection.InsertOne(ctx, token)
+	return err
+}
+
+func (r *MongoTokenRepository) GetToken(ctx context.Context, authToken string) (*models.Token, error) {
+	var token models.Token
+	err := r.collection.FindOne(ctx, bson.M{"authToken": authToken}).Decode(&token)
```

Comment:
Should we index token?

Reply:
No, already Indexed `clientId` and `scanRef`, which are used for querying tokens.
I see no benefit in indexing the actual token string.